packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -253,13 +253,24 @@ class GameWidgetState<T extends Game> extends State<GameWidget<T>> {
     } else {
       currentGame = widget.game!;
     }
-    currentGame.addGameStateListener(_onGameStateChange);
-    currentGame.lifecycleStateChange(
-      WidgetsBinding.instance.lifecycleState ?? AppLifecycleState.resumed,
-    );
+    initGameStateListener(currentGame, _onGameStateChange);
     _loaderFuture = null;
   }
 
+  /// Visible for testing for
+  /// https://github.com/flame-engine/flame/issues/2771.
+  @visibleForTesting
+  static void initGameStateListener(
```

Comment:
I still don't understand the need for a new function here, you already have access to do all that the function does from the test?
But even so, it feels like cheating the test by calling this manually?

Reply:
We can't use the `initCurrentGame` function since it's in a State object which isn't accessible with a backgrounded lifecycle state.
You're welcome to try some other solution but it's not as simple as the other tests.

Reply:
What I mean is that you can call this straight from the tests, instead of creating a new method:
```dart
currentGame.addGameStateListener(onGameStateChange); currentGame.lifecycleStateChange(AppLifecycleState.resumed);
```

Reply:
Oh that wouldn't do anything to prevent regressions though. It's also basically a duplicate of this existing test:
```dart
testWidgets(
  'game is paused when app is backgrounded',
  (tester) async {
    final game = FlameGame();

    await tester.pumpWidget(GameWidget(game: game));

    await game.toBeLoaded();
    await tester.pump();

    expect(game.paused, isFalse);
    WidgetsBinding.instance.handleAppLifecycleStateChanged(
      AppLifecycleState.paused,
    );
    expect(game.paused, isTrue);
    WidgetsBinding.instance.handleAppLifecycleStateChanged(
      AppLifecycleState.resumed,
    );
    expect(game.paused, isFalse);
  },
);
```

Reply:
I don't see how adding the new method would prevent regressions?

Reply:
Adding the new method allows us to test that we're using `AppLifecycleState.resumed` _not_ `WidgetsBinding.instance.lifecycleState` when the game is initialized.

Reply:
Ah alright, I missed that you were actually calling this from within the widget when reviewing on the phone. I guess without doing some much more complicated test we won't be able to test this better. Approved!